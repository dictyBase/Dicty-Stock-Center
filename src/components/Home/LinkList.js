// @flow
import React from "react"
import { Link } from "react-router-dom"
import useStyles from "./homeStyles"

type Props = {
  /** List of links in array form */
  list: Array<{
    name: string,
    to: string,
    routerAware: boolean,
  }>,
  /** Color of panel background, either gray or blue */
  bgColor: string,
}

/**
 * Generates a list of links based on a passed in array
 */

const LinkList = ({ list, bgColor }: Props) => {
  const classes = useStyles()

  return (
    <div className={bgColor === "gray" ? classes.panelGray : classes.panelBlue}>
      <ul className={classes.list}>
        {list.map((link, index) => (
          <li key={index}>
            {link.routerAware ? (
              <Link to={link.to} className={classes.link}>
                {link.name}
              </Link>
            ) : (
              <a href={link.to} className={classes.link}>
                {link.name}
              </a>
            )}
          </li>
        ))}
      </ul>
    </div>
  )
}

export default LinkList
