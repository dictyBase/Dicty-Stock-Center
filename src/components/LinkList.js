// @flow
import React, { Component } from "react"
import { Link } from "react-router-dom"
import { LinkedList } from "styles"

type Props = {
  list: Array<Object>,
  title: string
}

export default class LinkList extends Component<Props> {
  displayName = "list of links"

  render() {
    const { title, list } = this.props
    return (
      <LinkedList>
        {title && <h3>{title}</h3>}
        <ul>
          {list.map((link, index) => {
            return (
              <li key={index}>
                {link.routerAware ? (
                  <Link to={link.to}>{link.name}</Link>
                ) : (
                  <a href={link.to}>{link.name}</a>
                )}
              </li>
            )
          })}
        </ul>
      </LinkedList>
    )
  }
}
