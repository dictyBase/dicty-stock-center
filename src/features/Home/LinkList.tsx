import React from "react"
import { Link } from "react-router-dom"
import useStyles from "./homeStyles"

type LinkItem = {
  /** Name of link */
  name: string
  /** URL of link */
  to: string
  /** Indicates if link should use React Router */
  routerAware: boolean
  /** Indicates if link should open in new window */
  newWindow: boolean
}

type Props = {
  /** List of links in array form */
  list: Array<LinkItem>
  /** Color of panel background, either gray or blue */
  bgColor: string
}

const getLinkType = (link: LinkItem) => {
  if (link.routerAware) {
    return <Link to={link.to}>{link.name}</Link>
  }
  if (link.newWindow) {
    return (
      <a href={link.to} target="_blank" rel="noopener noreferrer">
        {link.name}
      </a>
    )
  }
  return <a href={link.to}>{link.name}</a>
}

/**
 * Generates a list of links based on a passed in array
 */

const LinkList = ({ list, bgColor }: Props) => {
  const classes = useStyles()

  // add simple header if the link list is for downloads panel
  const isDownloadPanel = list.some(
    (item) => item.name === "Phenotype Ontology",
  )

  return (
    <div className={bgColor === "gray" ? classes.panelGray : classes.panelBlue}>
      {isDownloadPanel && <h3>Download / View</h3>}
      <ul className={classes.list}>
        {list.map((link, index) => (
          <li key={index}>{getLinkType(link)}</li>
        ))}
      </ul>
    </div>
  )
}

export default LinkList
