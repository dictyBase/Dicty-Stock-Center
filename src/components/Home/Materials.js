// @flow
import React from "react"
import LinkList from "../LinkList"
import { withStyles } from "@material-ui/core/styles"
import styles from "./homeStyles"

type Props = {
  /** Material-UI styling */
  classes: {
    panelGray: string,
  },
}

const materials = [
  { name: "Strain Catalog", to: "/strains", routerAware: true },
  { name: "Plasmid Catalog", to: "/plasmids", routerAware: true },
  { name: "Bacterial Strains", to: "", routerAware: true },
  {
    name: "Additional Materials",
    to: "/information/additional-materials",
    routerAware: true,
  },
]

/**
 * Materials displays links to catalogs/materials on the homepage.
 */

const Materials = ({ classes }: Props) => (
  <div className={classes.panelGray}>
    <LinkList list={materials} />
  </div>
)

export default withStyles(styles)(Materials)
