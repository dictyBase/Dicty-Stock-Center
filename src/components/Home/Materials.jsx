// @flow
import React from "react"
import LinkList from "../LinkList"
import { PanelGray } from "styles"

const materials = [
  { name: "Strain Catalog", to: "/strains", routerAware: true },
  { name: "Plasmid Catalog", to: "/plasmids", routerAware: true },
  { name: "Bacterial Strains", to: "", routerAware: true },
  { name: "Other Materials", to: "", routerAware: true }
]

const Materials = () => {
  Materials.displayName = "links to stock catalogs"

  return (
    <PanelGray>
      <LinkList list={materials} />
    </PanelGray>
  )
}

export default Materials
