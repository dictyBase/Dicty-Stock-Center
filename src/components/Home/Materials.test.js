import React from "react"
import { mount } from "enzyme"
import Materials from "./Materials"
import LinkList from "../LinkList"
import { BrowserRouter } from "react-router-dom"

describe("Home/Materials", () => {
  const wrapper = mount(
    <BrowserRouter>
      <Materials />
    </BrowserRouter>,
  )
  describe("initial render", () => {
    it("always renders initial components", () => {
      expect(wrapper.find(LinkList)).toHaveLength(1)
    })
  })
})
