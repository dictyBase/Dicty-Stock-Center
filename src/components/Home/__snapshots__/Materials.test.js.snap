// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Materials 1`] = `
<div
  className="sc-dTdPqK dImFKM"
>
  <div
    className="sc-fzsDOv jrLCrg"
  >
    <ul>
      <li>
        <a
          href="/strains"
          onClick={[Function]}
        >
          Strain Catalog
        </a>
      </li>
      <li>
        <a
          href="/plasmids"
          onClick={[Function]}
        >
          Plasmid Catalog
        </a>
      </li>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Bacterial Strains
        </a>
      </li>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Other Materials
        </a>
      </li>
    </ul>
  </div>
</div>
`;
